docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+
+The node_modules directory should be absent. I'm not certain at the moment how to guard against a user-specified modulesFolder at build-time.
+
+Any additional configuration settings in the repository via [.yarnrc](https://classic.yarnpkg.com/en/docs/yarnrc) or .npmrc
+files should be ignored (at least initially) for the prefetch by specifying ```--no-default-rc``` for the yarn install command.
+
+We never previously honored user-defined settings with Cachito, but this can be investigated and implemented in Cachi2 in a follow-up. At minimum, the
+limitation should be documented.
+
+### SBOM Generation
+The Yarn 1.x CLI doesn't provide enough information to generate an SBOM, so the yarn.lock and package.json files will need to
+be parsed in order to gather the necessary data. The code used to process these files will need to be
+[imported from or inspired by Cachito](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py)
+and refactored/improved.
+
+#### Key Points
+ - The yarn.lock file can be parsed with [pyarn](https://github.com/containerbuildsystem/pyarn) like in cachito
+ - Workspaces will need to be [processed separately](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L95) from package.json since they do not appear in the yarn.lock file
+ - Dev dependencies can be [determined similiarly](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L140) to cachito
+ - We will need to ensure that there are no collisions in the names of the archive files being added to the offline mirror
+
+#### PURL Examples
+[NPM PURL specification](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#npm)
+
+##### Registry Package
+```txt
+pkg:npm/@optional-scope/package-name@1.0.0
+```
+
+##### URL Package
+```txt
+pkg:npm/package-name@1.0.0?checksum=sha512:checksum&download_url=https://example.com/package-name/0101010101010101010101010101010101010101.tar.gz
+```
+
+##### Git Package
+
+```txt
+pkg:npm/package-name@1.0.0?vcs_url=git%2Bhttps://example.com/namespace/package-name.git%0101010101010101010101010101010101010101
+```
+
+##### Workspace/File Package
+
+```txt
+pkg:npm/package-name@1.0.0?vcs_url=git%2Bhttps://example.com/namespace/package-name.git%0101010101010101010101010101010101010101#subpath
+```
+
+### Avoiding Arbitrary Code Execution
+Yarn 1.x appears to honor the ```--ignore-scripts``` option for the install command. It can also be configured via environment variable.
+Used https://github.com/chmeliik/js-lifecycle-scripts/tree/yarn to verify.
+
+### Prefetch
+The offline mirror in the request output directory will be populated by running the following Yarn command in the project root:
+
+    yarn install --non-interactive --frozen-lockfile --disable-pnp --no-default-rc --ignore-engines
```

Comment:
Are we potentially interested also in `--update-checksums` [1]? Since we're processing the repo in a temp environment we could see if the `yarn.lock` reports correct info that would not get updated during the actual install during hermetic build and report that fact (based on unstaged git changes) to the end user?
[1] https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-update-checksums

Reply:
Also, again, where does the `disable-pnp` option come from? Am I looking at the wrong docs?

Reply:
It's another CLI option that is not mentioned in their documentation: https://github.com/yarnpkg/yarn/blob/7cafa512a777048ce0b666080a24e80aae3d66a9/src/cli/index.js#L84

The PnP feature itself is though: https://classic.yarnpkg.com/en/docs/pnp

Reply:
Same as above, if we're dealing with internal-only option (at this point it doesn't make any sense to request a change in the docs from upstream) we should link them in the respective code base to clear up any potential confusion.

Reply:
> Just to make sure I understand: are you proposing we not fail the prefetch due to an incorrect checksum and only report it?

Disregard please, double checked and based on the description the option seems to be effectively mutually exclusive with `frozen-lockfile`